### PR TITLE
211-editor-permissions-fix

### LINKED ISSUE #211: User with editor role can modify posts of others
## Problem

Only users with `admin` roles should be able to modify/delete post of others

## Suggestion

- modify `operation/add-post`
- change required role in `/removed-post` end point middleware.

### PR DESCRIPTION
**Pull request by skydread1**
Closes #211
---

- only admin can delete posts
- only admin can edit posts from others

### CODE REVIEW

File: server/src/flybot/server/core/handler/operation.clj [hunk 1]
```diff
@@ -22,20 +22,32 @@
 
 ;;---------- Ops with effects ----------
 
+(defn admin?
+  [user]
+  (->> user :user/roles (map :role/name) (filter #(= :admin %)) seq))
+
```
Comment (flybot-nam-nguyenhoai) [inline server/src/flybot/server/core/handler/operation.clj:28]: `admin?` can probably be implemented using a pull pattern.

File: server/test/flybot/server/core/handler/operation_test.clj [hunk 2]
```diff
@@ -31,6 +31,22 @@
...
+               (sut/add-post (d/db db-conn) post-in)))))
+    (testing "User is admin and edits post of other so returns new post."
+      (let [post-in s/post-1
+            post-out (assoc s/post-1
+                            :post/author s/alice-user
+                            :post/last-editor s/bob-user)]
+        (is (= {:response post-out
+                :effects {:db {:payload [post-in]}}}
+               (sut/add-post (d/db db-conn) post-in)))))
+    (testing "User is not admin and tries tp edit post of other so returns error map."
```
Comment (flybot-nam-nguyenhoai) [inline server/test/flybot/server/core/handler/operation_test.clj:43]: Typo in `tries tp`; should be `tries to`.

File: server/src/flybot/server/core/handler/operation.clj (hunk 1 above)
Comment (skydread1) [reply server/src/flybot/server/core/handler/operation.clj:28]: Good suggestion, what about:

```clojure
(defn admin?
  [user]
  (->> user
       ((pull/qfn {:user/roles [{(list :role/name :when #(= :admin %)) '?} '?roles]} ?roles))
       (not-every? empty?)))
```

Comment (flybot-nam-nguyenhoai) [reply server/src/flybot/server/core/handler/operation.clj:28]: I was thinking
```clojure
(->> user
     ((pull/qfn
       {:user/roles [{:role/name '?} '?roles]}
        (some #{{:role/name :admin}} ?roles))))
```
but perhaps it's too long to be worth rewriting.

Comment (skydread1) [reply server/src/flybot/server/core/handler/operation.clj:28]: Yes, let's keep the original code.

Plus `pull/with-data-schema` triggers some unwanted validation against `v/api-schema` in that case.

**Pull request merged**